.github/workflows/ts-build.yml

```diff
@@ -0,0 +1,26 @@
+name: Typescript Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+    - 'typescript-demo/**'
+
+jobs:
+  build:
+    env:
+      shouldPushImage: ${{ github.ref == 'refs/head/main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
```

Comment:
Just a remark for my own personal culture, what's the point of using QEMU here ?

Reply:
I don't really know to be honest. It's part of all the examples in Github's documentation, so I assume it is required in setting up the Docker build tools

Reply:
I suspect this is getting around Docker in Docker.